types/state.go

```diff
@@ -25,6 +27,9 @@ type State struct {
 	LastBlockHeight uint64
 	LastBlockTime   time.Time
 
+	// LastHeaderHash is the hash of the header of the last block
```

Comment:
While this is correct, won't it break syncing for chains already in testnet?

Reply:
just tested on eden, it cant sync when i start current main, then switch to this branch. we should check if the value is empty if so then skip verification